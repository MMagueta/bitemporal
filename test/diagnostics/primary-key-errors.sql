\set QUIET 'on'
\pset footer off

drop schema if exists pk cascade;
create schema pk;

\echo 'bitemporal.primary_key_errors - [ok]'
\pset title 'pk.ok'
create table pk.ok
    ( id           int not null
    , id2          int not null
    , valid_period int4range not null
    , primary key(id, id2, valid_period) );

select * from bitemporal.primary_key_errors('pk.ok');

\echo 'bitemporal.primary_key_errors - [missing primary key]'
\pset title 'pk.missing'
create table pk.missing
    ( id int not null );

select * from bitemporal.primary_key_errors('pk.missing');

\echo 'bitemporal.primary_key_errors - [missing valid time]'
\pset title 'pk.missing_valid_time'
create table pk.missing_valid_time
    ( id int not null
    , primary key(id) );

select * from bitemporal.primary_key_errors('pk.missing_valid_time');
