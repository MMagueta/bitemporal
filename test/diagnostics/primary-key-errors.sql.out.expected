             pk.ok
 namespace | relation | message 
-----------+----------+---------

                 pk.missing
 namespace | relation |       message       
-----------+----------+---------------------
 pk        | missing  | Missing primary key

                              pk.missing_valid_time
 namespace |      relation      |                     message                     
-----------+--------------------+-------------------------------------------------
 pk        | missing_valid_time | Missing attribute "valid_period" on primary key

